src/hooks/useKeyboardShortcut/index.ts

```diff
@@ -0,0 +1,91 @@
+import { useCallback, useEffect, useRef } from "react";
+import { CharKey, ModifierKey } from "constants/keys";
+
+type UseKeyboardShortcutOptions = {
+  disabled?: boolean;
+  preventDefault?: boolean;
+  overrideIgnore?: boolean;
+};
+
+// Used to prevent shortcuts from being activated on input elements.
+const INPUT_ELEMENTS = ["INPUT", "TEXTAREA", "SELECT"];
+
+const useKeyboardShortcut = (
```

Comment:
I played around with this a bit what do you think about this?

I separated out the special modifier keys since those have their own fields in an event. It also makes it easy to check for them. This ensures we only run if every modifier key is pressed but we also limit it to only one character. 

```ts

const useKeyboardShortcut = (
  keys: Keys,
  cb: () => void,
  options: UseKeyboardShortcutOptions = {}
) => {
  if (!keys.modifierKeys?.length && !keys.charKey?.length) {
    throw new Error("Must provide at least one key");
  }
  const {
    disabled = false,
    preventDefault = true,
    overrideIgnore = false,
  } = options;
  const cbRef = useRef(cb);
  cbRef.current = cb;

  const handleKeydown = useCallback(
    (event: KeyboardEvent) => {
      let areModifierKeysPressed = true;
      if (keys.modifierKeys && keys.modifierKeys.length) {
        areModifierKeysPressed = getPressedModifierKeys(event).every(
          (modifierKey) => keys.modifierKeys?.includes(modifierKey)
        );
      }
      const isCharKeyPressed = keys.charKey === event.key;

      const shouldExecute =
        overrideIgnore ||
        !INPUT_ELEMENTS.includes((event.target as HTMLElement).tagName);

      console.log(areModifierKeysPressed, isCharKeyPressed, shouldExecute);
      if (areModifierKeysPressed && isCharKeyPressed) {
        if (shouldExecute) {
          // Prevent browser default behavior.
          if (preventDefault) {
            event.preventDefault();
          }
          cbRef.current();
        }
      }
    },
    [keys, preventDefault, overrideIgnore]
  );

  useEffect(() => {
    // There's no need to keep track of events if the component is disabled.
    if (disabled) {
      document.removeEventListener("keydown", handleKeydown);
    } else {
      document.addEventListener("keydown", handleKeydown);
    }
    return () => {
      document.removeEventListener("keydown", handleKeydown);
    };
  }, []);
};

// Function checks which modifier keys are pressed and returns an array of the pressed modifier keys
const getPressedModifierKeys = (event: KeyboardEvent) => {
  const pressedModifierKeys: ModifierKey[] = [];
  if (event.ctrlKey) {
    pressedModifierKeys.push(ModifierKey.Control);
  }
  if (event.altKey) {
    pressedModifierKeys.push(ModifierKey.Alt);
  }
  if (event.shiftKey) {
    pressedModifierKeys.push(ModifierKey.Shift);
  }
  return pressedModifierKeys;
};
```